src/cmd/services/m3coordinator/downsample/downsampler_test.go

```diff
@@ -1828,6 +1828,210 @@ func TestDownsamplerAggregationWithRulesConfigRollupRuleAndDropPolicyAndDropTime
 	testDownsamplerAggregation(t, testDownsampler)
 }
 
+func TestDownsamplerAggregationWithRulesConfigRollupRuleUntimedRollups(t *testing.T) {
+	t.Parallel()
+
+	gaugeMetrics := []testGaugeMetric{
+		{
+			tags: map[string]string{
+				nameTag:         "http_requests",
+				"app":           "nginx_edge",
+				"status_code":   "500",
+				"endpoint":      "/foo/bar",
+				"not_rolled_up": "not_rolled_up_value_1",
+			},
+			timedSamples: []testGaugeMetricTimedSample{
+				{value: 42},
+				{value: 12, offset: 2 * time.Second},
```

Comment:
interesting, so these will actually get rolled up into one data point even though the resolution is one second?

Reply:
Yeah, so because timestamp gets stamped on the agg side, what you pass in is ignored. 

I added another variety of test where we wait for the time before sending in the metric. That actually aggregates things into two datapoints.